llvm/lib/Transforms/Instrumentation/InstrProfiling.cpp

```diff
@@ -60,10 +60,7 @@ using namespace llvm;
 #define DEBUG_TYPE "instrprof"
 
 namespace llvm {
-cl::opt<bool>
-    DebugInfoCorrelate("debug-info-correlate",
-                       cl::desc("Use debug info to correlate profiles."),
-                       cl::init(false));
```

Comment:
Is it possible to make this an alias to `-profile-correlate=debug-info`? I do like the new flag better, but I want to make sure we don't break existing code that relies on this flag. We should give users a chance to migrate.

Reply:
I found cl::alias cannot be used to alias one option of another flag (e.g. -profile-correlate=debug-info). So, every place we uses ProfileCorrelate needs to have a check for DebugInfoCorrelate. I don't know how you update llvm toolchain, is it possible to switch to new flag at the time when updating the toolchain?

Reply:
I refactored some code to #69656 and kept this flag `-debug-info-correlate`.